components/datetime/src/timezone.rs

```diff
@@ -388,90 +388,128 @@ impl<'d> TimeZoneFormat<'d> {
         s
     }
 
-    /// Returns the time zone in generic location format as defined by the UTS-35 spec.
+    /// Writes the time zone in generic location format as defined by the UTS-35 spec.
     /// e.g. France Time
     /// https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology
-    pub(super) fn generic_location_format(
+    pub(super) fn generic_location_format<W: fmt::Write + ?Sized>(
         &self,
+        sink: &mut W,
         time_zone: &impl TimeZoneInput,
-    ) -> Option<Cow<'d, str>> {
+    ) -> Result<(), DateTimeFormatError> {
         // TODO(blocked on #277) Use formatter utility instead of replacing "{0}".
-        self.exemplar_city(time_zone)
-            .map(|location| Cow::Owned(self.zone_formats.region_format.replace("{0}", &location)))
+        sink.write_str(
+            &self
+                .exemplar_cities
+                .as_ref()
+                .and_then(|cities| time_zone.time_zone_id().and_then(|id| cities.get(id)))
```

Comment:
I vaguely remember that you made the `.get()` function here return a Cow, which sometimes gets built up or cloned.  Can you pass `w` one level deeper and skip returning the Cow, or else just verify that you don't perform any superfluous operations?

Reply:
Ah yes, I'm actually really glad you brought this up. 

Now that we're passing in a `Write` implementer directly into the format functions, rather than returning sometimes owned strings and sometimes borrowed strings, there is no need for any clones, and I've just pushed up a commit that removes it.